src/StrawberryShake/Client/src/Transport.WebSockets/ISocketClient.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using System.IO.Pipelines;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace StrawberryShake.Transport
+{
+    public interface ISocketClient
+        : IAsyncDisposable
+    {
+        Uri? Uri { get; set; }
+
+        string? Name { get; }
+
+        bool IsClosed { get; }
+
+        Task OpenAsync(
+            CancellationToken cancellationToken = default);
+
+        Task SendAsync(
+            ReadOnlyMemory<byte> message,
+            CancellationToken cancellationToken = default);
+
+        Task ReceiveAsync(
+            PipeWriter writer,
+            CancellationToken cancellationToken = default);
+
+        Task CloseAsync(
+            string message,
+            SocketCloseStatus closeStatus,
+            CancellationToken cancellationToken = default);
+
+        ISocketProtocol GetProtocol();
```

Comment:
The `SocketClient` does not longer expose the WebSocket directly, but rather has the abstract Read and Write on it

The SocketClient is also the one that knows the protocol after the handshake. GetProtocol returns the instance of the Protocol `Handler` (ISocketProtoclHandler might actually be a better name)